spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteTablePathSparkAction.java

```diff
@@ -404,10 +404,7 @@ private Set<Pair<String, String>> statsFileCopyPlan(
       Preconditions.checkArgument(
           before.fileSizeInBytes() == after.fileSizeInBytes(),
           "Before and after path rewrite, statistic file size should be same");
-      result.add(
-          Pair.of(
-              RewriteTablePathUtil.stagingPath(before.path(), sourcePrefix, stagingDir),
-              after.path()));
+      result.add(Pair.of(before.path(), after.path()));
```

Comment:
good catch I think we dont need to open and rewrite the content of the stats file, so it only need to copied from source to target, instead of from staging to target.